pam/main-exec.go

```diff
@@ -25,6 +27,19 @@ func init() {
 	// calling the dbus services from the process and so that the module PID
 	// check won't fail.
 	runtime.LockOSThread()
+
+	// Ask the kernel to send SIGTERM if the parent dies
+	if err := unix.Prctl(unix.PR_SET_PDEATHSIG, uintptr(syscall.SIGTERM), 0, 0, 0); err != nil {
+		log.Errorf(context.Background(), "failed to set PDEATHSIG: %v", err)
+		os.Exit(1)
```

Comment:
Yeah, I was wondering to use the same too... But maybe we can handle it also client-side, on the dbus side... Let me check that too.

However this is fine.

The changes on broker instead I'd hold since I had some refactoring there for another branch, and that side is quite delicate sadly (as we may cancel too early).